config/samples/cassandra.datastax.com_v1beta1_cassandradatacenter.yaml

```diff
@@ -1,7 +1,29 @@
 apiVersion: cassandra.datastax.com/v1beta1
 kind: CassandraDatacenter
 metadata:
-  name: cassandradatacenter-sample
+  name: dc2
```

Comment:
When in the OpenShift OperatorHub catalog we need to include the image coordinates which are UBI based at a minimum. Preferably we would point to the Red Hat container registry coordinates.

In either case we should update this sample with the appropriate keys to override the images.

* serverImage
* configBuilderImage
* systemLoggerImage

Additionally when running on OpenShift we must override the SecurityContext with an empty block as it creates per pod users and injects them.

```yaml
spec:
  dockerImageRunsAsCassandra: false
  podTemplateSpec:
    securityContext: {}
    containers:
      - name: cassandra
        securityContext: {}
```

Note the `spec.dockerImageRunsAsCassandra` is set to false. This is due to needing to override the spec with an empty block so that OpenShift can template out the values it needs. See [construct_podtemplatespec.go](https://github.com/k8ssandra/cass-operator/blob/ed4a19bdd2cad356467872c899c47afb758581e8/pkg/reconciliation/construct_podtemplatespec.go#L575) and [construct_statefulset.go](https://github.com/k8ssandra/cass-operator/blob/ed4a19bdd2cad356467872c899c47afb758581e8/pkg/reconciliation/construct_statefulset.go#L57-L64) for a context.

This feels a bit counterintuitive though. We should consider changes needed to make this a little less confusing. (@jsanda has a little bit of experience here given our recent call with a user)

Reply:
It's worth noting that whatever sample we include in the `ClusterServiceVersion` is used to prepopulate the `CassandraDatacenter` creation UI. Providing the appropriate values here will default them for users even if we don't have a UI element specified.

Reply:
The images section is something I'd still have to think about. I wanted this image to be deployable to OperatorHub also, in which case we probably want to fetch the images from hub.docker.com / quay.io. And then a certified version which would use the containerRegistry from Red Hat. I think the latter requires login with RH account to get the pull token?

Reply:
As for images, systemLoggerImage & configBuilderImage are always UBI based when cass-operator deploys them (1.8.0 and up). 

We can push modified ImageConfig for OLM that uses serverImages with UBI-ending, but do they exists? I don't see us building them in the management-api.

Reply:
As for DSE-images. Is there any need ever to not use the non-UBI images? Or should we deploy by default the UBI-images (like datastax/dse-server:6.8.17-ubi7) ?

Reply:
We can use images hosted outside of the RH registry, they just need to be UBI-based. I thought we were going to mirror Docker Hub to quay.io via Harbor, but it's more likely we should just push images to both locations from the GitHub actions. I can take a look at that or we can put it in the backlog (CC @jdonenine). 

It would be nice to have UBI images for all containers we're pushing. Is that something we're working on @jdonenine? (I think this question focuses on the `cassandra` container).

Reply:
> I can take a look at that or we can put it in the backlog (CC @jdonenine).

I certainly seems simpler to do the push from the actions we already have, or some sort of a copy as a follow up action.  If the requirement is that they are UBI images and not where they live, we can probably deal with this later and we can put it in the backlog for now.

> It would be nice to have UBI images for all containers we're pushing. Is that something we're working on @jdonenine? (I think this question focuses on the cassandra container).

We generally just went through this with with the older stuff that we patched up, although we didn't have to touch the cassandra container at that point.  It makes sense I think that we based everything on the UBI images generally, it will simplify for us here and also help us avoid vulnerability issues down the road.

What new images do we need at this point?

Reply:
IMHO we should strive to **always** use the UBI images.

Reply:
We don't have UBI images of Open source versions of Cassandra (the management-api doesn't build Cassandra itself)

Reply:
Right and that feels like a miss. It's not a blocker for this PR, but something we should consider. (@jdonenine we need to sync on this)